src/components/route-creation/CustomHeadersAdmin.tsx

```diff
@@ -0,0 +1,117 @@
+import {
+    Box,
+    Button,
+    Field,
+    IconDelete16,
+    InputField,
+    Table,
+    TableBody,
+    TableCell,
+    TableCellHead,
+    TableHead,
+    TableRow,
+} from '@dhis2/ui'
+import React, { useRef, useState } from 'react'
+import i18n from '../../locales'
+import { CustomHeaders } from '../../types/RouteInfo'
+import styles from './CustomHeadersAdmin.module.css'
+
+type CustomHeaderAdminProps = {
+    headers: CustomHeaders
+    setHeaders: (CustomHeaders) => void
+}
+const CustomHeadersAdmin = (props: CustomHeaderAdminProps) => {
+    const { headers = {}, setHeaders } = props
+    const [headerNameToAdd, setHeaderNameToAdd] = useState<string>()
+    const [headerValueToAdd, setHeaderValueToAdd] = useState<string>()
+    const inputRef = useRef(null)
+
+    const addHeader = () => {
+        setHeaderNameToAdd('')
+        setHeaderValueToAdd('')
+        setHeaders({ ...headers, [headerNameToAdd]: headerValueToAdd })
+    }
+
+    const removeHeader = (headerName) => {
+        const newHeaders = { ...headers }
+        delete newHeaders[headerName]
+        setHeaders(newHeaders)
+    }
+    return (
+        <Box width="400px">
```

Comment:
Question: Does the table inherit this width? Would it be better if it occupied more space?
<img width="1630" alt="headers-table" src="https://github.com/user-attachments/assets/1c0edc8f-3e45-47bf-a3a9-d5eac6049e2e" />

Reply:
I restricted it to 400px - similar to what the [new maintenance app](https://github.com/search?q=repo%3Adhis2%2Fmaintenance-app-beta%20400px&type=code) does - I think it looks too big on normal screen otherwise. But the whole app should get another UX review at some point, and we could tackle this.